source/common/router/header_formatter.cc

```diff
@@ -19,16 +21,26 @@ namespace Router {
 namespace {
 
 std::string formatUpstreamMetadataParseException(const std::string& params,
-                                                 const EnvoyException* cause = nullptr) {
-  std::string reason;
-  if (cause != nullptr) {
-    reason = fmt::format(", because {}", cause->what());
+                                                 const std::string& reason) {
+  std::string formatted_reason;
+  if (!reason.empty()) {
+    formatted_reason = fmt::format(", because {}", reason);
   }
 
   return fmt::format("Incorrect header configuration. Expected format "
                      "UPSTREAM_METADATA([\"namespace\", \"k\", ...]), actual format "
                      "UPSTREAM_METADATA{}{}",
-                     params, reason);
+                     params, formatted_reason);
+}
+
+std::string formatUpstreamMetadataParseException(const std::string& params,
+                                                 const EnvoyException* cause = nullptr) {
+  std::string reason;
+  if (cause != nullptr) {
+    reason = fmt::format("{}", cause->what());
```

Comment:
nit: You don't really need this format call. You can probably just use the tertiary operator directly inside the function call below.